llvm/include/llvm/SYCLLowerIR/LowerWGLocalMemory.h

```diff
@@ -0,0 +1,49 @@
+//===-- LowerWGLocalMemory.h - SYCL kernel local memory allocation pass ---===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This pass does the following for each allocate call to
+// __sycl_allocateLocalMemory(Size, Alignment) function at the kernel scope:
+// - inserts a global byte array of Size bytes with specified alignment
+//   in work group local address space.
```

Comment:
What does `global byte array on local address space` mean? Is it a program-scope array or kernel-scope array?

UPD: looking at the code below, it seems to be a global (i.e. program-scope) pointer to a local memory - is it correct?

Reply:
Yes, that is right. I updated description.